### PR TITLE
CC-21435: Configurable Product integration & migration guides

### PR DESCRIPTION
**Pull request by michbeck**
## PR Description

Ticket: https://spryker.atlassian.net/browse/CC-21435

Ticket for IG: https://spryker.atlassian.net/browse/CC-23445

Ticket for MG: https://spryker.atlassian.net/browse/CC-23444

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: _includes/pbc/all/upgrade-modules/upgrade-the-productconfiguration-module.md [hunk 1]
```diff
@@ -1,4 +1,32 @@
+## Upgrading from version 0.2.* to version 1.0.*
 
+*Estimated migration time: 5 minutes*
+
+`ProductConfiguration` v1.0.0 introduces the following backward incompatible changes:
+
+* Adjusted ProductConfigurationFacade::getProductConfigurationCollection() so it is now getting product configurations by ProductConfigurationCriteria instead of ProductConfigurationFilterTransfer.
+* Impacted ProductConfigurationCartToProductConfigurationFacadeBridge::getProductConfigurationCollection() with facade changes.
```
Comment (limeeugenia) [inline _includes/pbc/all/upgrade-modules/upgrade-the-productconfiguration-module.md:8]: This bridge is not from this module. What's the reason for it to be in this migration guide?

File: _includes/pbc/all/upgrade-modules/upgrade-the-productconfiguration-module.md [hunk 2]
```diff
@@ -1,4 +1,32 @@
...
+* Introduced spy_product_configuration.uuid field to allow unique identifier.
```
Comment (limeeugenia) [inline _includes/pbc/all/upgrade-modules/upgrade-the-productconfiguration-module.md:9]: Please mention the unique index introduction.

File: _includes/pbc/all/upgrade-modules/upgrade-the-productconfigurationsrestapi-module.md [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
+*Estimated migration time: 5 minutes*
 
-## Upgrading from version 0.1.* to version 0.2.*
+`ProductConfigurationsRestApi` v1.0.0 introduces the following backward incompatible changes:
 
-*Estimated migration time: 5 minutes*
+* Replaced ProductConfigurationStorageClientInterface::findProductConfigurationInstanceBySku() with ProductConfigurationStorageClientInterface::getProductConfigurationInstanceCollection().
 
-To upgrade the `ProductConfigurationsRestApi` module from version 0.1.* to version 0.2.*, do the following:
+To upgrade the `ProductConfigurationsRestApi` module from version 0.2.* to version 1.0.*, do the following:
```
Comment (limeeugenia) [inline _includes/pbc/all/upgrade-modules/upgrade-the-productconfigurationsrestapi-module.md:9]: I can see that in this document we replace the 0.1 to 0.2 migration with 0.2 to 1.0, while in the `_includes/pbc/all/upgrade-modules/upgrade-the-productconfiguration-module.md` we added a section keeping the old migration section. Please mmake these two changes consistent and either leave both or replace them in all the cases.

File: _includes/pbc/all/upgrade-modules/upgrade-the-productconfigurationstorage-module.md [hunk 4]
```diff
@@ -1,4 +1,26 @@
+## Upgrading from version 0.2.* to version 1.0.*
 
+*Estimated migration time: 5 minutes*
+
+`ProductConfigurationStorage` v1.0.0 introduces the following backward incompatible changes:
+
+* Replaced ProductConfigurationStorageClientInterface::findProductConfigurationInstanceBySku() with ProductConfigurationStorageClientInterface::getProductConfigurationInstanceCollection().
+* Replaced ProductConfigurationStorageClientInterface::findProductConfigurationInstancesIndexedBySku() with ProductConfigurationStorageClientInterface::getProductConfigurationInstanceCollection().
```
Comment (limeeugenia) [inline _includes/pbc/all/upgrade-modules/upgrade-the-productconfigurationstorage-module.md:8]: Please say which methods we removed and which added. This is the actual module adding the methods.

File: _includes/pbc/all/upgrade-modules/upgrade-the-salesproductconfiguration-module.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+```bash
+composer require "spryker/sales-product-configuration":"^1.0.0" --update-with-dependencies
+```
+
+
+Increased ProductConfiguration module version dependency.
+Removed ProductConfigurationOrderSaverPlugin.
+Removed ProductConfigurationOrderPostSavePlugin.
+Removed CheckoutExtension module version BC.
```
Comment (limeeugenia) [inline _includes/pbc/all/upgrade-modules/upgrade-the-salesproductconfiguration-module.md:17]: What are these items here? Things to check? things to do?

**Review by limeeugenia — COMMENTED** [review]
It looks like we miss some major migration guides: 13 majors with only 10 migration guides present.

**Comment by limeeugenia** [comment]
There are some majors in the shop too, so many more MG missing.

File: _includes/pbc/all/install-features/202212.1/install-the-product-configuration-feature.md [hunk 6]
```diff
@@ -0,0 +1,2228 @@
...
+|--------------|------------------|--------------------------------------------------------------------------------------------------------------------------------------|
+| Spryker Core | {{site.version}} | [Spryker Core feature integration](/docs/scos/dev/feature-integration-guides/{{site.version}}/spryker-core-feature-integration.html) |
+| Product      | {{site.version}} | [Product feature integration](/docs/scos/dev/feature-integration-guides/{{site.version}}/product-feature-integration.html)           |
+
+### 1) Install the required modules using Composer
+
+Install the required modules:
+
+```bash
+composer require "spryker-feature/product-labels:"{{site.version}}" --update-with-dependencies
```
[the remark this answers is not in the document]
Comment (michbeck) [reply _includes/pbc/all/install-features/202212.1/install-the-product-configuration-feature.md:1440]: note: product-configuration

**Comment by michbeck** [comment]
> There are some majors in the shop too, so many more MG missing.

As clarified it's ok that those with no relevant changes don't have MG.

**Comment by michbeck** [comment]
> It looks like we miss some major migration guides: 13 majors with only 10 migration guides present.

As clarified it's ok that those with no relevant changes don't have MG.

File: _includes/pbc/all/upgrade-modules/upgrade-the-salesproductconfiguration-module.md [hunk 7]
```diff
@@ -2,16 +2,16 @@
 
 *Estimated migration time: 5 minutes*
 
+`SalesProductConfiguration` v1.0.0 introduces the following backward incompatible changes:
+
+* Removed ProductConfigurationOrderSaverPlugin.
+* Removed ProductConfigurationOrderPostSavePlugin.
+* Removed CheckoutExtension module version.
```
Comment (limeeugenia) [inline _includes/pbc/all/upgrade-modules/upgrade-the-salesproductconfiguration-module.md:9]:
```suggestion
* Removed dependency on CheckoutExtension module.
```

File: _includes/pbc/all/upgrade-modules/upgrade-the-productconfiguration-module.md [hunk 8]
```diff
@@ -16,16 +16,11 @@ To upgrade the `ProductConfiguration` module from version 0.2.* to version 1.0.*
 composer require "spryker/product-configuration":"^1.0.0" --update-with-dependencies
 ```
 
-2. Generate transfer classes:
-
-```bash
-console transfer:generate
```
[the remark this answers is not in the document]
Comment (michbeck) [reply _includes/pbc/all/upgrade-modules/upgrade-the-productconfiguration-module.md]: Adjusted that it is executed before and after, but must say that i currently don't see the reason for that.

File: _includes/pbc/all/upgrade-modules/upgrade-the-salesproductconfiguration-module.md [hunk 9]
```diff
@@ -2,16 +2,16 @@
 
 *Estimated migration time: 5 minutes*
 
+`SalesProductConfiguration` v1.0.0 introduces the following backward incompatible changes:
+
+* Removed ProductConfigurationOrderSaverPlugin.
+* Removed ProductConfigurationOrderPostSavePlugin.
```
Comment (limeeugenia) [inline _includes/pbc/all/upgrade-modules/upgrade-the-salesproductconfiguration-module.md:8]: I was not able to find where we recommend setting up the replacement for these.

File: _includes/pbc/all/install-features/202212.1/install-the-product-configuration-feature.md [hunk 10]
```diff
@@ -0,0 +1,2206 @@
...
+|--------------|------------------|--------------------------------------------------------------------------------------------------------------------------------------|
+| Spryker Core | {{site.version}} | [Spryker Core feature integration](/docs/scos/dev/feature-integration-guides/{{site.version}}/spryker-core-feature-integration.html) |
+| Product      | {{site.version}} | [Product feature integration](/docs/scos/dev/feature-integration-guides/{{site.version}}/product-feature-integration.html)           |
+
+### 1) Install the required modules using Composer
+
+Install the required modules:
+
+```bash
+composer require "spryker-feature/product-configuration:"{{site.version}}" --update-with-dependencies
```
Comment (limeeugenia) [inline _includes/pbc/all/install-features/202212.1/install-the-product-configuration-feature.md:1419]: please require the existing package

File: _includes/pbc/all/install-features/202212.1/install-the-product-configuration-feature.md [hunk 11]
```diff
@@ -28,7 +28,7 @@ To start feature integration, integrate the required features:
 Install the required modules:
 
 ```bash
-composer require spryker-feature/configurable-product:"{{site.version}}" --update-with-dependencies
+composer require "spryker-feature/configurable-product":"^1.0.0" --update-with-dependencies
```
Comment (limeeugenia) [inline _includes/pbc/all/install-features/202212.1/install-the-product-configuration-feature.md:31]: Please leave the {{site.version}} in.

File: _includes/pbc/all/install-features/202212.1/install-the-product-configuration-feature.md [hunk 12]
```diff
@@ -1785,7 +1785,7 @@ Overview and install the necessary features before beginning the integration ste
 ### 1) Install the required modules using Composer
 
 ```bash
-composer require "spryker-shop/date-time-configurator-page-example:"dev-master" --update-with-dependencies
+composer require "spryker-shop/date-time-configurator-page-example":"dev-master" --update-with-dependencies
```
Comment (limeeugenia) [inline _includes/pbc/all/install-features/202212.1/install-the-product-configuration-feature.md:1788]: `dev-master` is not allowed in the IGs, please use the exact version.

File: _includes/pbc/all/install-features/202212.1/install-the-product-configuration-feature.md [hunk 13]
```diff
@@ -0,0 +1,2206 @@
...
+
+| NAME         | VERSION          | INTEGRATION GUIDE                                                                                                                    |
+|--------------|------------------|--------------------------------------------------------------------------------------------------------------------------------------|
+| Spryker Core | {{site.version}} | [Spryker Core feature integration](/docs/scos/dev/feature-integration-guides/{{site.version}}/spryker-core-feature-integration.html) |
+| Product      | {{site.version}} | [Product feature integration](/docs/scos/dev/feature-integration-guides/{{site.version}}/product-feature-integration.html)           |
+
+### 1) Install the required modules using Composer
+
+```bash
+composer require "spryker-shop/date-time-configurator-page-example":"^0.2.3" --update-with-dependencies
```
Comment (limeeugenia) [inline _includes/pbc/all/install-features/202212.1/install-the-product-configuration-feature.md:1788]: We are going to release a minor on the module. Please use 0.3.0 version.

**Review by limeeugenia — DISMISSED** [review]

**Review by KarinOf — DISMISSED** [review]

**Review by lenadoc — APPROVED** [review]

**Pull request merged**